src/write/file.rs

```diff
@@ -119,8 +126,13 @@ impl<W: Write> FileWriter<W> {
         Ok(())
     }
 
-    /// Writes the footer of the parquet file. Returns the total size of the file.
-    pub fn end(&mut self, key_value_metadata: Option<Vec<KeyValue>>) -> Result<u64> {
+    /// Writes the footer of the parquet file. Returns the total size of the file and the
+    /// underlying writer.
+    pub fn end(mut self, key_value_metadata: Option<Vec<KeyValue>>) -> Result<(u64, W)> {
```

Comment:
I would avoid this - not every consumer has full ownership of the writer that it can destroy it. This is related to this comment: https://github.com/jorgecarleitao/arrow2/issues/980#issuecomment-1118662819 - a user may want to close the file without yielding ownership of the writer.

What do you think about a design based on [this](https://github.com/jorgecarleitao/arrow2/blob/main/src/io/ipc/write/writer.rs#L19)? There we safeguard against double `end` [like this](https://github.com/jorgecarleitao/arrow2/blob/main/src/io/ipc/write/writer.rs#L170).

Reply:
I have no issue with storing the state; it just means that it will no longer align with the async version?

Should the async `end` method be changed not to consume itself if holding onto the writer is desired? It could also have `.into_inner()` added.

https://github.com/jorgecarleitao/parquet2/blob/47edd88ab831da74c5e17d7b0e483ddc243f0f6a/src/write/stream.rs#L117-L123

Reply:
Ahh, I see - yes, there is a misalignment, well spotted. Yes, I think we should also avoid it in that case for the same reasons (if you agree with them!).

Reply:
The last commit should address this